### ISSUE TITLE
Scheduler blackout time period

### ISSUE THREAD

**Issue by boxysean** [apache/airflow#1191]
I have the need for a scheduler blackout time period in Airflow.

My team, which uses Airflow, has been asked to not query one of my company's data sources between midnight and 7 AM. When we launch big backfills on this data source, it would be nice to have the Scheduler not schedule some TaskInstances during the blackout hours.

We (@r39132 and @ledsusop) brainstormed a few ideas on gitter on how to do this...

(1) Put more state/logic in the TaskInstance and Scheduler like this:

```
my_task = PythonOperator(
    task_id='my_task',
    python_callable=my_command_that_access_the_datasource,
    provide_context=True,
    dag=dag,
    blackout=my_blackout_logic_for_the_datasource # <---
    )
```

where `my_blackout_logic` is some function I provide that the scheduler calls to determine whether or not it is the blackout period.

(2) Pause DAGs on nightly basis. We can do this outside of Airflow with cron or Jenkins already. Maybe this functionality gets incorporated into DAGs?

(3) @ledsusop seems to have some ideas on how to build DAGs that are sensitive to time.

(4) Extend my PythonOperator and implement logic for the blackout time period.

Solution (1) is ideal for me because only some TaskInstances in my DAG should be delayed. I'm probably going to go with (2), make my Jenkins pause/unpause my offending DAG during the blackout period for now.

Onwards. :tada:

**Comment by bolkedebruin** [comment]
@boxysean can you please create a jira for this? thanks!

**Comment by r39132** [comment]
I'd say this is done via the `airflow pause_dag <dag_id>` & `airflow unpause_dag <dag_id>` CLIs added and released in 1.7.0. It can be called by an external cron. If someone would like to add this functionality into the UI as a separate pause-unpause schedule, we would welcome the PR. I'm closing this issue for now, but it can be reopened as a Jira should such a PR be issued in the future.

**Comment by boxysean** [comment]
Yeah I agree that cron can do the pause / unpause.

The connection pools in Airflow are very nice because they allow for the limitation of the number of connections to be made and are some easy way for us to control the connections. This ask is for a feature that gives that capability within the Airflow UI and system, as I could imagine blackouts could be something that enterprise users might have to deal with (at least I do).

Created the issue, [AIRFLOW-81](https://issues.apache.org/jira/browse/AIRFLOW-81).
